Trustroots/trustroots
# Refactor eslint out from Gulpfile

#### Proposed Changes

* Run eslint directly from npm
* Consolidate Angular and other eslint configs to one; should make it easier to phase out Angular files during rewrite.
* Add `.eslintignore` file with some problematic directories (we'll clean up some of these in later PRs) https://eslint.org/docs/user-guide/configuring#eslintignore

#### Testing Instructions

1. Run `npm start`, `npm run test:server:watch` or `npm run test:client:watch`
2. Modify a file, you should see linting error:
    ```
    trustroots/karma.conf.js (1/0)
      ✖  21:57  Unexpected trailing comma  comma-dangle
    
    ✖ 1 error (9:13:59 PM)
    ```
3. Fix the file, you should see linting success:
    ```
    ✓ Clean (9:14:01 PM)
    ```

Confirm that `npm run lint` works, too.

Confirm that changing anything in Angular files (e.g. `const test = window.test;`) brings up warning about using Angular `$window` service, while that change would work in React files.
